src/usePopperMarginModifiers.js

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useRef, useMemo } from 'react';
+import hasClass from 'dom-helpers/hasClass';
+
+function getMargins(element) {
+  const styles = getComputedStyle(element);
+
+  const top = parseFloat(styles.marginTop) || 0;
+  const right = parseFloat(styles.marginRight) || 0;
+  const bottom = parseFloat(styles.marginBottom) || 0;
+  const left = parseFloat(styles.marginLeft) || 0;
+  return { top, right, bottom, left };
+}
+
+export default function usePopperMarginModifiers() {
+  const overlayRef = useRef(null);
+  const margins = useRef(null);
+
+  return [
+    useCallback((overlay) => {
+      if (
+        !overlay ||
+        !(hasClass(overlay, 'popover') || hasClass(overlay, 'dropdown-menu'))
+      )
+        return;
+
+      margins.current = getMargins(overlay);
+      overlay.style.margin = 0;
+      overlayRef.current = overlay;
```

Comment:
is layout effect too late? it seems like it might be nice to remove the ref at the end... though i guess it doesn't matter?

maybe leave a comment on why we're doing this, at least, somewhere in this module